services/aws-s3.service.ts

```diff
@@ -48,37 +56,47 @@ export class AppAwsS3Service {
     }
   }
 
-  public setCredentials(options) {
+  public setCredentials(options: {
+    credentials: { accessKeyId: string; secretKey: string };
+    region?: string;
+    apiVersion?: string;
+  }) {
     const creds = options.credentials;
-    const awsCreds = new AWS.Credentials({
+    const awsCreds = {
       accessKeyId: creds.accessKeyId,
       secretAccessKey: creds.secretKey,
+    };
+    this.s3Client = new S3Client({
+      apiVersion: options.apiVersion ?? this.defaultApiVersion,
+      credentials: awsCreds,
+      region: options.region ?? this.defaultRegion,
     });
-    this.s3Client = new AWS.S3({ credentials: awsCreds });
+    return this.s3Client;
   }
 
-  public getSignedUrlInternal(options) {
+  public async getSignedUrlInternal(options) {
```

Comment:
I believe this is supposed to be a private function? Can you double check this?

Reply:
yeah I couldn't find it being used externally anywhere, made private 👍